bundles/admin/admin-layereditor/view/Flyout.js

```diff
@@ -41,46 +41,53 @@ export class LayerEditorFlyout extends ExtraFlyout {
         this.update();
     }
     setLayer (layer) {
-        this.service.initLayerState(layer);
-        this.update();
+        if (!layer) {
+            this.service.resetLayer();
+        } else {
+            this.service.fetchLayer(layer.getId());
+        }
     }
     setDataProviders (dataProviders) {
         this.dataProviders = dataProviders;
     }
     setMapLayerGroups (mapLayerGroups) {
         this.mapLayerGroups = mapLayerGroups;
     }
-
     update () {
-        const me = this;
         const el = this.getElement();
         if (!el) {
             return;
         }
+        let uiCode = this.getEditorUI();
+        if (this.service.isLoading()) {
+            uiCode = <Spin>{ uiCode }</Spin>;
```

Comment:
Move this logic to `LayerWizard`.

Reply:
Good idea. There's a lot of stuff that should probably be moved around still but since we are prioritizing on getting the new layerlisting ready instead of the admin functionality it's good to get the "pending" work towards the admin merged even if it's not ready since the version in develop isn't ready either.

Reply:
Added TODO